Source/WebCore/platform/graphics/gstreamer/eme/WebKitOpenCDMDecryptorGStreamer.cpp

```diff
@@ -143,7 +143,7 @@ static void webKitMediaOpenCDMDecryptorFinalize(GObject* object)
     GST_CALL_PARENT(G_OBJECT_CLASS, finalize, (object));
 }
 
-static SessionResult webKitMediaOpenCDMDecryptorResetSessionFromInitDataIfNeeded(WebKitMediaCommonEncryptionDecrypt* self, const WebCore::InitData& initData)
+static SessionResult webKitMediaOpenCDMDecryptorResetSessionFromKeyIdIfNeeded(WebKitMediaCommonEncryptionDecrypt* self, Ref<WebCore::SharedBuffer>&& keyId)
```

Comment:
The use you make of the key id here does not take ownership of it, it was only uses it to ask for the session. I think you should make this `const SharedBuffer&` instead of `RefPtr<SharedBuffer>&&`. Of course, same goes for all the function calls up the onion until `webkitMediaCommonEncryptionDecryptSinkEventHandler` that checkes the key id and passes it down.